sdk/dns/Azure.ResourceManager.Dns/src/Customization/DnsRecordData.cs

```diff
@@ -142,11 +143,25 @@ public string Cname
         public DnsSoaRecordInfo DnsSoaRecordInfo { get; set; }
         /// <summary> The list of CAA records in the record set. </summary>
         public IList<DnsCaaRecordInfo> DnsCaaRecords { get; }
-         /// <summary> The list of DS records in the record set. </summary>
+        /// <summary> The list of DS records in the record set. </summary>
         public IList<DnsDSRecordInfo> DnsDSRecords { get; }
         /// <summary> The list of TLSA records in the record set. </summary>
         public IList<DnsTlsaRecordInfo> DnsTlsaRecords { get; }
         /// <summary> The list of NAPTR records in the record set. </summary>
         public IList<DnsNaptrRecordInfo> DnsNaptrRecords { get; }
+        /// <summary> The DnsRecordType in the record set. </summary>
+        public DnsRecordType RecordType
+        {
+            get
+            {
+                var resourceTypeString = base.ResourceType.Type.Split('/').Where(part => !string.IsNullOrEmpty(part)).LastOrDefault();
+                resourceTypeString = resourceTypeString == "AAAA" ? "Aaaa" : resourceTypeString;
+                resourceTypeString = resourceTypeString == "CNAME" ? "Cname" : resourceTypeString;
+                resourceTypeString = resourceTypeString == "TLSA" ? "Tlsa" : resourceTypeString;
+                resourceTypeString = resourceTypeString == "NAPTR" ? "Naptr" : resourceTypeString;
+                DnsRecordType recordType = (DnsRecordType)Enum.Parse(typeof(DnsRecordType), resourceTypeString);
```

Comment:
how about we just add the ignore casing here?
```suggestion
                DnsRecordType recordType = Enum.Parse<DnsRecordType>(typeof(DnsRecordType), resourceTypeString, true);
```

Reply:
use `DnsRecordType recordType = DnsRecordTypeExtensions.ToDnsRecordType(resourceTypeString);`  could be better